docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+
+|dimension
+|Vectors dimension
+|Y
+|null
+
+|metric
+|Distance metric used to calculate the distance between two vectors.
+See table Available distance metrics.
+|Y
+|null
+
+|max-degree
+|The maximum number of neighbors per node is calculated as max-degree * 2
+|Y
+|16
+
+|ef-construction
+|The size of the search queue to use when finding nearest neighbors.
+|Y
+|100
+
+|use-deduplication
+|The option for vector deduplication.
+When disabled, a new node will be created for each vector, even if it is the same as an existing one.
+|true
+
+|===
+
+.Available distance metrics
+[cols="2,2,2",options="header"]
+|===
+|Name|Description| Score definition
+
+|EUCLIDEAN
+|euclidean distance
+|1 / (1 + squareDistance(v1, v2))
+
+|COSINE
+|cosine of the angle between the vectors
+|(1 + cos(v1, v2)) / 2
+
+|DOT
+|dot product of the vectors
+|(1 + dotProduct(v1, v2)) / 2
+
+|===
+
+WARNING: The recommended method for computing cosine similarity is to normalize all vectors to unit length and utilize the DOT metric instead.
+
+
+Configuration example:
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    <vector-collection name="vector-collection-name">
+        <indexes>
+            <index name="index-1">
+                <dimension>2</dimension>
+                <metric>DOT</metric>
+            </index>
+            <index name="index-2">
+                <dimension>2</dimension>
+                <metric>DOT</metric>
+                <max-degree>10</max-degree>
+                <ef-construction>20</ef-construction>
+                <use-deduplication>false</use-deduplication>
+            </index>
+        </indexes>
+    </vector-collection>
+</hazelcast>
+----
+--
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  vector-collection:
+    vector-collection-name:
+      indexes:
+        - name: index-1
+          dimension: 2
+          metric: DOT
+        - name: index-2
+          dimension: 2
+          metric: DOT
+          max-degree: 10
+          ef-construction: 20
+          use-deduplication: false
+----
+--
+Java::
++
+--
+[source,java]
+----
+Config config = new Config();
+VectorCollectionConfig collectionConfig = new VectorCollectionConfig("vector-collection-name")
+    .addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-1")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+    ).addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-2")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+                .setMaxDegree(10)
+                .setEfConstruction(20)
+                .setUseDeduplication(false)
+    );
+config.addVectorCollectionConfig(collectionConfig);
+----
+--
+====
+
+== Manage collection
+
+=== Create collection
+
+To create a vector collection, use the static method `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)` of the `VectorCollection` class.
```

Comment:
`create a vector collection` / `get a vector collection` are not completely on the point. It does not matter if the collection exists. It matters if the _configuration_ exists. And you can use `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)` many times provided that `collectionConfig` is always the same.

Reply:
64e5181268054046bf02460e682f405097c2dea6